addons/account/models/account_payment.py

```diff
@@ -403,12 +404,19 @@ def _compute_payment_method_line_fields(self):
             to_exclude = self._get_payment_method_codes_to_exclude()
             if to_exclude:
                 pay.available_payment_method_line_ids = pay.available_payment_method_line_ids.filtered(lambda x: x.code not in to_exclude)
-            if pay.payment_method_line_id.id not in pay.available_payment_method_line_ids.ids:
-                # In some cases, we could be linked to a payment method line that has been unlinked from the journal.
-                # In such cases, we want to show it on the payment.
-                pay.hide_payment_method_line = False
+
+    @api.depends('payment_type')
+    def _compute_available_journal_ids(self):
+        """
+        Get all journals having at least one payment method for inbound/outbound depending on the payment_type.
+        Done with filtered because the domain does not work as expected on these fields.
```

Comment:
We don't care about the implementation in the docstring; if it is worth mentioning, put it as a comment at the right place.

The docstring should onnly contain a functional description, the input, the output, and maybe the side effects (input modified) or the complexity in some cases.